lib/homework1.ex

```diff
@@ -1,18 +1,73 @@
 defmodule Homework1 do
   @moduledoc """
-  Documentation for `Homework1`.
+  This module implements the simple calculation function.
+  You can use the `Homework1.calculate/1` function to add, subtract, multiply and divide.
   """
 
   @doc """
-  Hello world.
+  Calculate the result of an expression:
 
   ## Examples
-
-      iex> Homework1.hello()
-      :world
+      iex> Homework1.calculate "1 + 1"
+      2.0
+      iex> Homework1.calculate "12 - 76"
+      -64.0
+      iex> Homework1.calculate "11 * 9"
+      99.0
+      iex> Homework1.calculate "11 / 2"
+      5.5
 
   """
-  def hello do
-    :world
+
+  @spec calculate(String.t()) :: number()
+  def calculate(expression) when is_binary(expression) do
+    expression
+    |> validate
+    |> String.replace(" ", "")
+    |> Float.parse()
+    |> compute()
+  end
+
+  @spec validate(String.t()) :: String.t()
+  defp validate(expression) do
+    has_add_or_subtract = String.split(expression, ~r/\+|\-/, parts: 2) |> length > 1
+    has_multiply_or_divide = String.split(expression, ~r/\*|\//, parts: 2) |> length > 1
+
+    if has_add_or_subtract and has_multiply_or_divide do
```

Comment:
Sounds like the dilemma of [Philip II of Macedon](https://en.wikipedia.org/wiki/Divide_and_rule) :)